lib/AST/DiagnosticConsumer.cpp

```diff
@@ -116,6 +116,21 @@ FileSpecificDiagnosticConsumer::consumerForLocation(SourceManager &SM,
   // FileSpecificDiagnosticConsumer to be set up before the source files are
   // actually loaded.
   if (ConsumersOrderedByRange.empty()) {
+
+    // It's possible to get here while a bridging header PCH is being
+    // attached-to, if there's some sort of AST-reader warning or error, which
+    // happens before CompilerInstance::setUpInputs(), at which point _no_
+    // source buffers are loaded in yet. In that case we return nullptr, rather
+    // than trying to build a nonsensical map (and actually crashing since we
+    // can't find buffers for the inputs).
+    if (!(SubConsumers.empty() ||
+          SM.getIDForBufferIdentifier(SubConsumers.begin()->first).hasValue())) {
```

Comment:
Honestly if `SubConsumers` is empty we're in trouble anyway. We may as well assert that so that we don't drop diagnostics on the floor. (This is good too because I find this if-condition hard to read.)